apps/heft/src/utilities/WatchFileSystemAdapter.ts

```diff
@@ -0,0 +1,215 @@
+    callback?: ReaddirDirentCallback | ReaddirStringCallback
+  ) => {
+    filePath = path.normalize(filePath);
+    // Default to no options, which will return a string callback
+    let options: IReaddirOptions | undefined;
+    if (typeof optionsOrCallback === 'object') {
+      options = optionsOrCallback;
+    } else if (typeof optionsOrCallback === 'function') {
+      callback = optionsOrCallback;
+    }
+
+    if (options?.withFileTypes) {
+      fs.readdir(filePath, options, (err: NodeJS.ErrnoException | null, entries: fs.Dirent[]) => {
+        if (err) {
+          this._missing.set(filePath, Date.now());
+        } else {
+          this._contexts.set(filePath, Date.now());
+        }
+        (callback as ReaddirDirentCallback)(err, entries);
+      });
+    } else {
+      fs.readdir(filePath, (err: NodeJS.ErrnoException | null, entries: string[]) => {
+        if (err) {
+          this._missing.set(filePath, Date.now());
+        } else {
+          this._contexts.set(filePath, Date.now());
+        }
+        (callback as ReaddirStringCallback)(err, entries);
+      });
+    }
+  };
+
+  /** { @inheritdoc fs.lstat } */
+  public lstat: StatAsynchronousMethod = (filePath: string, callback: StatCallback): void => {
+    filePath = path.normalize(filePath);
+    fs.lstat(filePath, (err: NodeJS.ErrnoException | null, stats: fs.Stats) => {
+      if (err) {
+        this._missing.set(filePath, Date.now());
+      } else {
+        this._files.set(filePath, stats.mtimeMs || stats.ctimeMs || Date.now());
+      }
+      callback(err, stats);
+    });
+  };
+
+  /** { @inheritdoc fs.lstatSync } */
+  public lstatSync: StatSynchronousMethod = (filePath: string): fs.Stats => {
+    filePath = path.normalize(filePath);
+    try {
+      const stats: fs.Stats = fs.lstatSync(filePath);
+      this._files.set(filePath, stats.mtimeMs || stats.ctimeMs || Date.now());
+      return stats;
+    } catch (err) {
+      this._missing.set(filePath, Date.now());
+      throw err;
+    }
+  };
+
+  /** { @inheritdoc fs.stat } */
+  public stat: StatAsynchronousMethod = (filePath: string, callback: StatCallback): void => {
+    filePath = path.normalize(filePath);
+    fs.stat(filePath, (err: NodeJS.ErrnoException | null, stats: fs.Stats) => {
+      if (err) {
+        this._missing.set(filePath, Date.now());
+      } else {
+        this._files.set(filePath, stats.mtimeMs || stats.ctimeMs || Date.now());
+      }
+      callback(err, stats);
+    });
+  };
+
+  /** { @inheritdoc fs.statSync } */
+  public statSync: StatSynchronousMethod = (filePath: string) => {
+    filePath = path.normalize(filePath);
+    try {
+      const stats: fs.Stats = fs.statSync(filePath);
+      this._files.set(filePath, stats.mtimeMs || stats.ctimeMs || Date.now());
+      return stats;
+    } catch (err) {
+      this._missing.set(filePath, Date.now());
+      throw err;
+    }
+  };
+
+  public reset(): void {
```

Comment:
Maybe call this `pause` or `restartWatcher`. The implication of `reset` is that it reinitializes this object.

Reply:
`prepare`?